### ISSUE TITLE
VMSS cluster not being created with acs-engine 0.16.2

### ISSUE THREAD

**Issue by shayansarkar** [Azure/acs-engine#2885]
**Is this a request for help?**:
Yes
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
Issue
---

**What version of acs-engine?**:
0.16.2
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes

**What happened**:
I tried to create a k8s cluster with VMSS and I received an error saying 

``` Error returned by LoadContainerService: VirtualMachineScaleSets are not supported with Kubernetes ```

**What you expected to happen**:
From my understanding with k8s 1.10 and acs-engine 0.16.2 I should be able to have a VMSS based k8s cluster.

**How to reproduce it** (as minimally and precisely as possible):

I created a test json file and ran acs-engine generate vmss.json and received the error saying VirtualMachineScaleSets are not supported.  

```
{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.10"
    },
    "masterProfile": {
      "count": 1,
      "dnsPrefix": "vmss-test",
      "vmSize": "Standard_D2_v2"
    },
    "agentPoolProfiles": [
      {
        "name": "agentpool1",
        "count": 3,
        "vmSize": "Standard_D2_v2",
        "availabilityProfile": "VirtualMachineScaleSets",
        "storageProfile": "ManagedDisks"
      }
    ],
    "linuxProfile": {
        "adminUsername": "k8suser",
        "ssh": {
            "publicKeys": [{
                "keyData": ""
            }]
        }
    },
    "servicePrincipalProfile": {
        "clientId": "",
        "secret": ""
    }
  }
}
```

**Anything else we need to know**:

**Comment by feiskyer** [comment]
Seems vmss support is not included in v0.16.2, have you tried master build?

**Comment by shayansarkar** [comment]
If I build it from master, I get a different error

```
Error returned by LoadContainerService: OrchestratorProfile is not able to be rationalized, check supported Release or Version
```

But from what I understand, VMSS should be the default k8s option in the current version of acs-engine.  So not sure why I would need to run the master version.  In the availabiliytProfile section of the documentation it says that if you are trying to run k8s 1.10 or later, it should default to a VMSS

https://github.com/Azure/acs-engine/blob/master/docs/clusterdefinition.md

This is the acs-engine version output when running from master
```
Version: canary
GitCommit: e44e42f0
GitTreeState: clean
```

**Comment by feiskyer** [comment]
> GitCommit: e44e42f0

The commit doesn't seem from master. Could you fetch latest commits and try again?

**Comment by shayansarkar** [comment]
Oh, my fault.  For some reason when I checked out the master branch with go, it didn't pull the latest.  Not sure how that happened.  

Anyways, with the master build, it did generate the template although that behavior seems to be different from what the documentation implies.  It seems like VMSS should be the default for any new cluster.

**Comment by CecileRobertMichon** [comment]
@shayansarkar with what apimodel did you notice VMSS not being default?

**Comment by shayansarkar** [comment]
I just took this VMSS example and added my service principal and cert at the bottom.  Then I ran it with the 0.16.2 version of acs-engine and got the error about VirtualMachineScaleSets are not supported with Kubernetes.  I can generate everything with the master version of acs-engine, but it seemed like the documentation said that 0.16.2 should have worked.

https://github.com/Azure/acs-engine/blob/master/examples/kubernetes-vmss/kubernetes.json
